### PR TITLE
gcworker: fix resource leak.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -688,6 +688,13 @@ func (w *GCWorker) loadValueFromSysTable(key string, s tidb.Session) (string, er
 	goCtx := goctx.Background()
 	stmt := fmt.Sprintf(`SELECT (variable_value) FROM mysql.tidb WHERE variable_name='%s' FOR UPDATE`, key)
 	rs, err := s.Execute(goCtx, stmt)
+	if len(rs) > 0 {
+		defer func() {
```
Comment (coocood) [inline store/tikv/gcworker/gc_worker.go]: How about `defer terror.Call(rs[0].Close)`

File: store/tikv/sql_fail_test.go [hunk 2]
```diff
@@ -54,6 +55,13 @@ func (s *testSQLSuite) TestFailBusyServerCop(c *C) {
 	go func() {
 		defer wg.Done()
 		rs, err := session.Execute(goctx.Background(), `SELECT variable_value FROM mysql.tidb WHERE variable_name="bootstrapped"`)
+		if len(rs) > 0 {
+			defer func() {
```
Comment (coocood) [inline store/tikv/sql_fail_test.go]: ditto

**Comment by shenli** [comment]
@coocood PTAL

File: ddl/util/util.go [hunk 3]
```diff
@@ -45,6 +46,9 @@ func (t DelRangeTask) Range() ([]byte, []byte) {
 func LoadDeleteRanges(ctx context.Context, safePoint uint64) (ranges []DelRangeTask, _ error) {
 	sql := fmt.Sprintf(loadDeleteRangeSQL, safePoint)
 	rss, err := ctx.(sqlexec.SQLExecutor).Execute(goctx.TODO(), sql)
+	if len(rss) > 0 {
+		defer terror.Call(rss[0].Close)
```
Comment (zz-jason) [inline ddl/util/util.go]: maybe this is more safe:
```go
defer func() {
    for _, rs := range rss {
        terror.Call(rs.Close)
    }
}()
```

Comment (coocood) [reply ddl/util/util.go]: We can make sure we only execute one statement.

**Comment by coocood** [comment]
LGTM

File: ddl/util/util.go (hunk 3 above)
Comment (disksing) [reply ddl/util/util.go]: rss can contain at most 1 recordSet, the code block below also depends on this constraint.

Comment (zz-jason) [reply ddl/util/util.go]: OK

**Pull request merged**